src/frontend/JasmineGraphFrontEnd.cpp

```diff
@@ -1314,1497 +1420,126 @@ bool JasmineGraphFrontEnd::isGraphActive(std::string graphID, SQLiteDBInterface
-            std::vector<string> partitionVec = partitionMap.find(workerID)->second;
-            partitionVec.push_back(partitionId);
-        }
-
-        frontend_logger.log("###FRONTEND### Getting Triangle Count : Host " + host + " Server Port " +
-                            serverPort + " PartitionId " + partitionId, "info");
-
-    }
-
-    for (auto &&futureCall:remoteCopyRes) {
-        futureCall.wait();
-    }
-
-    for (int i = 0; i < workerListSize; i++) {
-        Utils::worker currentWorker = workerList.at(i);
-        string host = currentWorker.hostname;
-        string workerID = currentWorker.workerID;
-        string partitionId;
-
-        std::vector<string> partitionList = partitionMap[workerID];
-
-        std::vector<string>::iterator partitionIterator;
-
-        for (partitionIterator = partitionList.begin(); partitionIterator != partitionList.end(); ++partitionIterator) {
-            partitionCount++;
-            int workerPort = atoi(string(currentWorker.port).c_str());
-            int workerDataPort = atoi(string(currentWorker.dataPort).c_str());
-            frontend_logger.log("====>PORT:" + std::to_string(workerPort), "info");
-
-            partitionId = *partitionIterator;
-            intermRes.push_back(
-                    std::async(std::launch::async, JasmineGraphFrontEnd::getTriangleCount, atoi(graphId.c_str()), host,
-                               workerPort, workerDataPort, atoi(partitionId.c_str()), masterIP, uniqueId, isCompositeAggregation));
-        }
-    }
-
-    for (auto &&futureCall:intermRes) {
-        result += futureCall.get();
-    }
-
-    if (!isCompositeAggregation) {
-        long aggregatedTriangleCount = JasmineGraphFrontEnd::aggregateCentralStoreTriangles(sqlite, graphId,masterIP);
-        result += aggregatedTriangleCount;
-        frontend_logger.log("###FRONTEND### Getting Triangle Count : Completed: Triangles " + to_string(result),
-                            "info");
-    }
-
-    auto end = chrono::high_resolution_clock::now();
-    auto dur = end - begin;
-    auto msDuration = std::chrono::duration_cast<std::chrono::milliseconds>(dur).count();
-
-    std::string durationString = std::to_string(msDuration);
-
-    if (canCalibrate) {
-        Utils::updateSLAInformation(perfSqlite, graphId, partitionCount, msDuration, TRIANGLES, Conts::SLA_CATEGORY::LATENCY);
-    }
-
 
-    triangleTree.clear();
-    combinationWorkerMap.clear();
-
-    return result;
+void JasmineGraphFrontEnd::getAndUpdateUploadTime(std::string graphID, SQLiteDBInterface sqlite) {
+    struct tm tm;
+    vector<vector<pair<string, string>>> uploadStartFinishTimes = sqlite.runSelect(
+            "SELECT upload_start_time,upload_end_time FROM graph WHERE idgraph = '" + graphID + "'");
+    string startTime = uploadStartFinishTimes[0][0].second;
+    string endTime = uploadStartFinishTimes[0][1].second;
+    string sTime = startTime.substr(startTime.size() - 14, startTime.size() - 5);
```

Comment:
What are these 14 and 5. Can we have them in constants?

Reply:
These were there in the original file. I didn't do any changes to these areas Dr Miyuru.

Reply:
Ok